File: C++ Programming/Competitive Programming/Coding Challenges 3/program2.cpp
/*
2. Accept number of rows and number of columns from user and display
below pattern.
Input : iRow = 4  iCol = 4
Output :2 4 6 8
        1 3 5 7
        2 4 6 8
        1 3 5 7             
*/
#include<iostream>
using namespace std;

void Pattern(int iRow,int iCol)
{
    int i =1;
    int j= 1;
    int iCnt = 1;

    while (i<=iRow)
    {
        j=1;

        if (i % 2 != 0)
        {
            iCnt = 2;
        }
        else
        {
            iCnt =1;
        }

        while (j<=iCol)
        {
            cout<<iCnt<<"\t";
            iCnt= iCnt+2;

            j++;
        }
        cout<<"\n";
         i++;
    }
    
}

int main()
{
int iValue1=0, iValue2=0,iRet=0;

cout<<"Enter the number of Rows : ";
cin>>iValue1;

cout<<"Enter the number of Columns : ";
cin>>iValue2;

Pattern(iValue1,iValue2);

return 0;
}

File: Java Programming/Competitive Programming/Coding Challenges 16/program2.java
/*
2. Write a program which accept range from user and display all even numbers in
between that range.
Input : 23 35
Output : 24 26 28 30 32 34
Input : 10 18
Output : 10 12 14 16 18
Input : 10 10
Output : 10
Input : -10 2
Output : -10 -8 -6 -4 -2 0 2
Input : 90 18
Output : Invalid range
*/

import java.util.Scanner;

class program2
{
    static void RangeDisplayEven(int iStart, int iEnd)
    {
        if (iStart > iEnd)
        {
            return;
        }

        if (iStart % 2 == 0)
        {
            System.out.print(iStart + "\t");
        }

        RangeDisplayEven(iStart + 1, iEnd);
    }

    public static void main(String[] args)
    {
        Scanner sobj = new Scanner(System.in);

        int iValue1 = 0;
        int iValue2 = 0;

        System.out.print("Enter starting point: ");
        iValue1 = sobj.nextInt();

        System.out.print("Enter ending point: ");
        iValue2 = sobj.nextInt();

        if (iValue1 > iValue2)
        {
            System.out.print("Invalid Range");
        }
        else
        {
            RangeDisplayEven(iValue1, iValue2);
        }

        sobj.close();
    }
}